saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java

```diff
@@ -192,9 +194,26 @@ public Saml2MessageBinding getAssertionConsumerServiceBinding() {
 	 * relying party's &lt;SPSSODescriptor&gt;.
 	 * @return the SingleLogoutService Binding
 	 * @since 5.6
+	 * @deprecated use {@link #getSingleLogoutServiceBindings()}
 	 */
+	@Deprecated
```

Comment:
I hesitate to mark this as deprecated. 

While I agree that the naming isn't ideal, it's quite common in SAML to have a situation where the spec supports multiple values but the majority of the time there is only one. Consider, for example, `Saml2AuthenticatedPrincipal#getFirstAttribute`. I imagine that it will be quite common that an application only supports one binding and now have that extra boilerplate of `get(0)` to do. If they are going to need to do it anyway, and we are already doing it, we might as well keep the method around. 

In that case, the JavaDoc would change to say that it returns the first one in the list.

I'm open to being convinced otherwise if you are strongly in favor of deprecation.

Reply:
I was mainly trying to preserve the contract. Would you be opposed to throwing an exception if there are multiple? My reasoning is that the term first would imply ordering which then would have to carry to other classes such as the SamlMetadataResolver.

Reply:
Sure, that sounds reasonable.